src/xbot/common/math/PID.java

```diff
@@ -23,6 +28,37 @@ public void reset()
     {
         m_prevError = 0;
         m_totalError = 0;
+        errorIsSmall = false;
+        derivativeIsSmall = false;
+    }
+    
+    /**
+     * Set how close the error can be before it is considered
+     * "on-target."
+     *            
+     * This is in the same units as your current and goal values.
+     */
+    public void setErrorTolerance(double errorTolerance) {
+        setTolerances(errorTolerance, this.derivativeTolerance);
```

Comment:
The PID class currently is just a bunch of math - it has no visibility into properties.